### ISSUE TITLE
PartialOrd implementation for rask_engine::math::{Vec2, Vec3}

### ISSUE THREAD

**Issue by konsumlamm** [squirrel-labs/ratatosk#53]
The implementation of `PartialOrd` for `Vec2` and `Vec3` is not trivial. The current definition is that `v < w` if `v[i] < w[i]` for all components i. We should either clearly document this behaviour or make a separate function for implementing this kind of comparison.

**Comment by NatrixAeria** [comment]
Would it be reasonable to implement `Iter` (and `FromIterator` for `.collect()`) for of `math::VecX`?
The only usage of `PartialOrd` since yet is:
```rust
impl Collide<Vec2> for AABox {
    fn collides(&self, other: &Vec2) -> bool {
        self.pos < *other && *other < self.pos + self.size
    }
}
```
So my approach would be to do `vec1.zip(vec2).all(f32::le)` instead of `vec1 < vec2`.
Also this would be a more consistent implementation for a AABox. The current `collides` function returns `true` if the given point is in the AABox or touches an edge of it. But if the point is at `aabox.pos` or `aabox.pos + aabox.size` then it returns `false`.

**Comment by TrueDoctor** [comment]
sounds reasonable and I can't think of anything that would object an implementation of `Iter`

**Comment by NatrixAeria** [comment]
fixed by 0dad915576c36bee4f26eec0f519d0b7eb79aea7
